src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -320,23 +321,75 @@ public void removeWorkspaceRole(
   @Traced
   public String syncWorkspacePolicy(
       UUID workspaceId, WsmIamRole role, AuthenticatedUserRequest userReq) {
+    String group =
+        syncPolicyOnObject(
+            SamConstants.SAM_WORKSPACE_RESOURCE, workspaceId.toString(), role.toSamRole(), userReq);
+    logger.info(
+        "Synced role {} to google group {} in workspace {}", role.toSamRole(), group, workspaceId);
+    return group;
+  }
+
+  /**
+   * Wrapper around Sam client to sync a Sam policy on a controlled resource to a google group and
+   * return the email of that group.
+   *
+   * <p>This should only be called for controlled resources which require permissions granted to
+   * individual users, i.e. private or application-controlled resources. All other cases are handled
+   * by the permissions that workspace-level roles inherit on resources, and do not use the policies
+   * synced by this function.
+   *
+   * <p>This operation in Sam is idempotent, so we don't worry about calling this multiple times.
+   *
+   * @param resource The resource to sync a binding for
+   * @param role The policy to sync in Sam
+   * @param userReq User authentication
+   * @return
+   */
+  @Traced
+  public String syncPrivateResourcePolicy(
+      ControlledResource resource,
+      ControlledResourceIamRole role,
+      AuthenticatedUserRequest userReq) {
+    // TODO: in the future, this function will also be called for application managed resources,
+    //  including app-shared. This check should be modified appropriately.
+    if (resource.getAccessScope() != AccessScopeType.ACCESS_SCOPE_PRIVATE) {
+      throw new InternalLogicException(
+          "syncPrivateResourcePolicy should not be called for shared resources!");
+    }
+    String group =
+        syncPolicyOnObject(
+            SamControlledResourceNames.get(resource.getAccessScope(), resource.getManagedBy()),
+            resource.getResourceId().toString(),
+            role.toSamRole(),
+            userReq);
+    logger.info(
+        "Synced role {} to google group {} for resource {}",
+        role.toSamRole(),
+        group,
+        resource.getResourceId());
+    return group;
+  }
+
+  /**
+   * Common implementation for syncing a policy to a Google group on an object in Sam.
+   *
+   * @param resourceTypeName The type of the Sam object, as configured with Sam.
```

Comment:
nit: 'Sam object' or 'Sam resource'?

Reply:
In Sam terms, "resource". inside WSM, that's overloaded so I went with `syncPolicyOnObject` for the method - cleaned up params/javadocs for consistency though.